src/zenml/orchestrators/utils.py

```diff
@@ -50,6 +51,35 @@ def create_tfx_pipeline(
     )
 
 
+def is_cached_step(execution_info: data_types.ExecutionInfo) -> bool:
+    """Returns the caching status of a tfx step.
+
+    Args:
+        execution_info: The execution info of a tfx step.
+
+    Returns:
+        The caching status of a tfx step as a boolean value.
+    """
+    repository = Repository()
+    metadata_store = repository.get_stack(
```

Comment:
What if the stack changes in the middle? i.e. a pipeline is long-running (.e.g two hours). While its running user changes stack.

We should probably not get the active stack here but the stack used by the pipeline. This will be in exec_properties I believe when @bcdurak branch is merged

Reply:
Waiting on the branch to be merged to work on this.

Reply:
Managed to implement this, though it's a bit messy due to how it's stored in the `ExecutionInfo` object. Just need to get the test amended to this new pattern which is proving a bit fiddly, but it seems doable.